pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -172,4 +174,38 @@ func toSelectableFields(claim *resource.ResourceClaim) fields.Set {
 
 // dropDisabledFields removes fields which are covered by a feature gate.
 func dropDisabledFields(newClaim, oldClaim *resource.ResourceClaim) {
+	dropDisabledDRAAdminAccessFields(newClaim, oldClaim)
+}
+
+func dropDisabledDRAAdminAccessFields(newClaim, oldClaim *resource.ResourceClaim) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRAAdminAccess) {
+		// No need to drop anything.
+		return
+	}
+
+	for i := range newClaim.Spec.Devices.Requests {
```

Comment:
Are you 100% sure that you can associate old and new requests based on index?

Reply:
When assuming that specs are immutable, yes. But this code shouldn't make that assumption.

Anyway, I simplified it ("in use anywhere in the claim").